saleor/graphql/checkout/mutations.py

```diff
@@ -1035,6 +1053,176 @@ def perform_mutation(
         return CheckoutShippingMethodUpdate(checkout=checkout)
 
 
+class CheckoutDeliveryMethodUpdate(BaseMutation):
+    checkout = graphene.Field(Checkout, description="An updated checkout.")
+
+    class Arguments:
+        token = UUID(description="Checkout token.", required=False)
+        delivery_method_id = graphene.ID(
+            description="Delivery Method ID (`Warehouse` ID or `ShippingMethod` ID).",
+            required=False,
+        )
+
+    class Meta:
+        description = (
+            "Updates the delivery method (shipping method or pick up point)"
+            "of the checkout."
+        )
+        error_type_class = CheckoutError
+        error_type_field = "checkout_error"
+
+    @classmethod
+    def perform_on_shipping_method(
+        cls, info, shipping_method_id, checkout_info, lines, checkout, manager
+    ):
+        shipping_method = cls.get_node_or_error(
+            info,
+            shipping_method_id,
+            only_type=ShippingMethod,
+            field="shipping_method_id",
+            qs=shipping_models.ShippingMethod.objects.prefetch_related(
+                "postal_code_rules"
+            ),
+        )
+
+        cls._check_delivery_method(
+            checkout_info, lines, shipping_method=shipping_method, collection_point=None
+        )
+
+        checkout_delivery_method_update = cls._update_delivery_method(
+            manager,
+            checkout,
+            shipping_method=shipping_method,
+            collection_point=None,
+        )
+        recalculate_checkout_discount(
+            manager, checkout_info, lines, info.context.discounts
+        )
+        return checkout_delivery_method_update
+
+    @classmethod
+    def perform_on_collection_point(
+        cls, info, collection_point_id, checkout_info, lines, checkout, manager
+    ):
+        collection_point = cls.get_node_or_error(
+            info,
+            collection_point_id,
+            only_type=Warehouse,
+            field="collection_point_id",
```

Comment:
We don't have this field in input. In `field` we should use only names from mutation input. This field is used to return errors in case of an invalid id.

Reply:
```suggestion
            field="delivery_method_id",
```

Reply:
Done.